pkg/k8s/client_test.go

```diff
@@ -197,11 +198,16 @@ func (s *K8sSuite) Test_runHeartbeat(c *C) {
 		},
 	)
 
-	select {
-	case <-time.After(5 * time.Millisecond):
-		c.Error("Heartbeat should have closed all connections")
-	case <-called:
-	}
+	err := testutils.WaitUntil(func() bool {
```

Comment:
Please add a comment on why the select in `WaitUntil` is needed. Otherwise, it's not clear why a standard `time.After()` timeout in a single select isn't sufficient.